source/DSCResources/AddsDomainController/AddsDomainController.schema.psm1

```diff
@@ -103,16 +111,38 @@ configuration AddsDomainController
         $depOn = '[Script]RemoveProtectFromAccidentalDeletionBeforeDcPromo'
     }
 
-    ADDomainController 'DomainControllerAllProperties' {
-        DomainName                    = $DomainName
-        Credential                    = $Credential
-        SafeModeAdministratorPassword = $SafeModeAdministratorPassword
-        DatabasePath                  = $DatabasePath
-        LogPath                       = $LogPath
-        SysvolPath                    = $SysvolPath
-        SiteName                      = $SiteName
-        ReadOnlyReplica               = $IsReadOnlyReplica
-        IsGlobalCatalog               = $IsGlobalCatalog
-        DependsOn                     = $depOn
+    if ($IsReadOnlyReplica -eq $false)
+    {
+        ADDomainController 'DomainControllerAllProperties'
+        {
+
+            DomainName                    = $DomainName
+            Credential                    = $Credential
+            SafeModeAdministratorPassword = $SafeModeAdministratorPassword
+            DatabasePath                  = $DatabasePath
+            LogPath                       = $LogPath
+            SysvolPath                    = $SysvolPath
+            SiteName                      = $SiteName
+            ReadOnlyReplica               = $IsReadOnlyReplica
+            IsGlobalCatalog               = $IsGlobalCatalog
+            DependsOn                     = $depOn
+        }
+    }elseif ($IsReadOnlyReplica -eq $true)
```

Comment:
Just a trifle, but can you fix the formatting. In VSCode, `ALT + Shift + F` should do.